cmd/uid_provider/client/client.go

```diff
@@ -317,12 +341,22 @@ func hasPermission(ctx context.Context, rdb *redis.ClusterClient, uid, pwd strin
 		return fmt.Errorf("invalid password for user %v", uid)
 	}
 
-	if rootPerm, ok := requestor.Permissions[""]; ok {
-		if utils.Includes(rootPerm, string(auth.CreateToken)) {
-			return nil
+	for newUserPermissionScope, newUserPermissions := range newUser.Permissions {
+		isAllowed := false
+		for requestorPermissionScope, requestorPermissions := range requestor.Permissions {
+			if isPermissionAllowed(newUserPermissionScope, newUserPermissions, requestorPermissionScope, requestorPermissions, isCreation) {
+				isAllowed = true
+				break
+			}
+		}
+
+		if !isAllowed {
+			return ierrors.NewError().Forbidden().Message("You are not allowed to create/delete/update a user with those permissions").Build()
```

Comment:
Capitalized error message. Also, I'd change it to something like: "not allowed to create/delete/update a user with current permissions"

Reply:
Changed it.